pkg/storegateway/stats.go

```diff
@@ -76,8 +77,17 @@ type queryStats struct {
 	streamingSeriesAmbientTime time.Duration
 }
 
+// blockSourceAndLevel encapsulate a block's thanos source and compaction level
+type blockSourceAndLevel struct {
+	source string
```

Comment:
can we keep using the `block.SourceType` type here? it will make tracing back the possible values for this easier